### ISSUE TITLE
[Accessibility]when accessibilityRole set to header on android, no heading suffix pronounced in talkback

### ISSUE THREAD

**Issue by yangweigbh** [facebook/react-native#22440]
## Environment
 React Native Environment Info:
    System:
      OS: Windows 10
      CPU: x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
      Memory: 6.31 GB / 15.93 GB
    Binaries:
      npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.2.0.0 AI-181.5540.7.32.5014246

    "react": "16.5.0",
    "react-native": "0.57.1"

## Description
on android, setting accessibilityRole to "header" for view component does not have any effect when talkback turn on, they will neither be pronounced with heading suffix nor can navigate by heading (when talkback turn on, "navigation by heading" can be turn on by swiping up)

## Reproducible Step:

sample code:
```
<Text accessibilityRole="header">Welcome to React Native!</Text>
<Text accessibilityRole="header">To get started, edit App.js</Text>
<Text accessibilityRole="header">blah,blah</Text>
```
1. turn on talkback in setting -> accessibility -> talkback
2. launch react native app
3. tap on ` <Text>`, there is no "heading" suffix pronounced
4. swiping up to switch "navigation mode" to "heading", and the `<Text>` component can not be found as "heading"

**Comment by yangweigbh** [comment]
regarding your implementation in 
[AccessibilityDelegateUtil.java](https://github.com/facebook/react-native/blob/c5186aeb461a6eafa582b0acba1b3cebfb83550b/ReactAndroid/src/main/java/com/facebook/react/uimanager/AccessibilityDelegateUtil.java)

I think the correct implementation for heading should using the  `AccessibilityNodeInfo.CollectionItemInfo` api

```
ViewCompat.setAccessibilityDelegate(
        view,
        new AccessibilityDelegateCompat() {
          @Override
          public void onInitializeAccessibilityNodeInfo(
            View host, AccessibilityNodeInfoCompat info) {
            super.onInitializeAccessibilityNodeInfo(host, info);
            
             final AccessibilityNodeInfo.CollectionItemInfo itemInfo =
                AccessibilityNodeInfo.CollectionItemInfo.obtain(
                        0, 1, 0, 1,
                        true);
			info.setCollectionItemInfo(itemInfo);
          }
        });
    }
```

**Comment by bartolkaruza** [comment]
@yangweigbh would you be interested in opening a PR with this improvement? You can build the Android sources for React Native using these [steps](https://facebook.github.io/react-native/docs/building-from-source). That way you could at least test if your proposed change will actually have the right effect.

**Comment by yangweigbh** [comment]
@bartolkaruza PR submited, please review 

https://github.com/facebook/react-native/pull/22447
